protocol/statetracker/spec_updater.go

```diff
@@ -0,0 +1,65 @@
+package statetracker
+
+import (
+	"context"
+
+	"github.com/lavanet/lava/protocol/chainlib"
+	"github.com/lavanet/lava/utils"
+	spectypes "github.com/lavanet/lava/x/spec/types"
+)
+
+const (
+	CallbackKeyForSpecUpdate = "spec-update"
+)
+
+type SpecGetter interface {
+	GetSpec(ctx context.Context, chainID string) (*spectypes.Spec, error)
+}
+
+type SpecUpdater struct {
+	eventTracker     *EventTracker
+	chainParser      chainlib.ChainParser
+	chainId          string
+	apiInterface     string
+	specGetter       SpecGetter
+	blockLastUpdated uint64
+}
+
+func NewSpecUpdater(chainId string, chainParser chainlib.ChainParser, specGetter SpecGetter, eventTracker *EventTracker, apiInterface string) *SpecUpdater {
+	return &SpecUpdater{chainId: chainId, chainParser: chainParser, specGetter: specGetter, eventTracker: eventTracker, apiInterface: apiInterface}
+}
+
+func (su *SpecUpdater) UpdaterKey() string {
+	utils.LavaFormatDebug("UpdaterKey: " + CallbackKeyForSpecUpdate + su.chainId + su.apiInterface)
+	return CallbackKeyForSpecUpdate + su.chainId + su.apiInterface
+}
+
```

Comment:
missing RegisterForSpecUpdates, if you support more than one, you need this
also you need locks for it like in the other updaters in the fix PR